docs/src/main/sphinx/admin/properties-http-client.rst

```diff
@@ -0,0 +1,312 @@
+======================
+HTTP client properties
+======================
+
+``http-client.https.hostname-verification``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-boolean`
+* **Default value:** ``true``
+
+Verifies that the server hostname matches the server certificate.
+
+``http-client.http2.enabled``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-boolean`
+* **Default value:**
+
+.. No default value?
+
+Enables the HTTP/2 transport.
+
+``http-client.connect-timeout``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-duration`
+* **Minimum value:** ``0ms``
+* **Default value:** ``5s``
+
+``http-client.request-timeout``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-duration`
+* **Minimum value:** ``0ms``
+* **Default value:** ``5m``
+
+``http-client.idle-timeout``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-duration`
+* **Minimum value:** ``0ms``
+* **Default value:** ``1m``
+
+``http-client.max-connections``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-integer`
+* **Minimum value:** ``1M``
+* **Default value:** ``200``
+
+``http-client.max-connections-per-server``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-integer`
+* **Minimum value:** ``1M``
+* **Default value:** ``20``
+
+``http-client.max-requests-queued-per-destination``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-integer`
+* **Minimum value:** ``1M``
+* **Default value:** ``1024``
+
+``http-client.max-content-length``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-data-size`
+* **Default value:** ``16MB``
+
+``http-client.request-buffer-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-data-size`
+* **Maximum value:** ``32MB``
+* **Default value:** ``4KB``
+
+``http-client.response-buffer-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-data-size`
+* **Maximum value:** ``32MB``
+* **Default value:** ``16KB``
+
+``http-client.socks-proxy``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** 
+* **Default value:**
+
+..
+    Host and port?
+
+``http-client.http-proxy``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:**
+* **Default value:**
+
+..
+    Host and port?
+
+``http-client.http-proxy.secure``
```

Comment:
And again the same as above. HTTP vs. HTTPS, this being specifying whether or not it's HTTPS. No default value because I guess they want you to be explicit with it.